.clang-format

```
@@ -0,0 +1,95 @@
+# OSARA: Open Source Accessibility for the REAPER Application
+# .clang-format
+# Author: James Teh <jamie@jantrid.net>
+# Copyright 2023 Leonard de Ruijter
+# License: GNU General Public License version 2.0
+
+# Style based on Google's C++ style guide.
+BasedOnStyle: Google
+# Offset for access modifiers (public, protected, private).
+AccessModifierOffset: 0
+# Align code after an open bracket with block indentation.
+AlignAfterOpenBracket: BlockIndent
```

I think we should use DontAlign here. Right now, a lot of code puts arguments on the same line as the function call, only breaking the line when more would make the line too long. Using DontAlign would result in less churn.

```suggestion
AlignAfterOpenBracket: DontAlign
```

---

Ug, this causes long if statements to have their final line indented to the same level as the inner block. That's... annoying. We could fix this with ContinuationIndentWidth: 4. That affects function calls as well, but that's probably okay.

---

I have investigated that. Personally I really dislike the style where the closing paren after a list of function parameters isn't at the same level as the function name. For that, we really need BlockIndent.

---

But this is already what we have. There are a lot of places where the code does things like this:

```
    foo(bar, baz,
        blahblah);
```

Codifying this means we get less churn.

---

This works for function calls, but for function definitions, it is either:

```
int foo(
	int bar, int bazz) {
```

or

```
int foo(int bar,
	int bazz) {
```

Wherea's I'd probably prefer:

```
int foo(int bar,
	int bazz
) {
```

As we use in several places.

---

If I understand correctly, DontAlign should mean we get this if we set continuation indentation to 4:

```
int foo(int bar,
		int baz) {
	...
}
```

So it's possible to distinguish between the function parameters and the body because the parameters are further indented than the body is. That said, I agree it does look particularly weird for function definitions, and it doesn't look like we can differentiate between definitions and calls. That's unfortunate because breaking calls up like that does mean a lot of extraneous nearly empty lines. Ah well.